src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHostBuilderExtensions.cs

```diff
@@ -40,11 +41,19 @@ public static IHostBuilder UseSystemd(this IHostBuilder hostBuilder)
                         options.FormatterName = ConsoleFormatterNames.Systemd;
                     });
 
-                    services.AddSingleton<ISystemdNotifier, SystemdNotifier>();
-                    services.AddSingleton<IHostLifetime, SystemdLifetime>();
+                    ConfigureLifetime(services);
                 });
             }
             return hostBuilder;
         }
+
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "ServiceDescriptor has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```

Comment:
```suggestion
            Justification = "AddSingleton has RequiresUnreferencedCode because Service and Implementation types could be generic with mis-matching trimming annotations. " +
```